src/core/thread/data_poll_manager.cpp

```diff
@@ -422,7 +422,7 @@ void DataPollManager::HandlePollTimer(Timer &aTimer)
 uint32_t DataPollManager::GetDefaultPollPeriod(void) const
 {
     return TimerMilli::SecToMsec(GetNetif().GetMle().GetTimeout()) -
-           static_cast<uint32_t>(kRetxPollPeriod) * kMaxPollRetxAttempts;
+           static_cast<uint32_t>(kRetxPollPeriod) * Mle::kFailedChildTransmissions;
```

Comment:
Good catch!
Well I would prefer to update `kMaxPollRetxAttempts` to be the same as `kMaxChildKeepAliveAttempts`(i.e. 4) as 

1. it is where `kMniTimeout` derives from `kMinTimeout = (((kMaxChildKeepAliveAttempts + 1) * kUnicastRetransmissionDelay) / 1000), ///< Minimum timeout(s)`. And

2. keep data_poll_manager less dependent on constants defined in other files.

Thoughts?

Reply:
Defined configuration in `openthread-core-default-config.h` and updated the expression to calculate kMniTimeout.